File: Sistema_Papeleria_AURUM_Frontend/src/app/clases/inventario.ts
export class Inventario {
  categoria: string;
  fechaIngreso: Date;
  nombreProducto: string;
  idProducto: number;
  cantidadProducto: number;

  constructor(
    categoria: string,
    fechaIngreso: Date,
    nombreProducto: string,
    idProducto: number,
    cantidadProducto: number
  ) {
    this.categoria = categoria;
    this.fechaIngreso = fechaIngreso;
    this.nombreProducto = nombreProducto;
    this.idProducto = idProducto;
    this.cantidadProducto = cantidadProducto;
  }
}

File: Sistema_Papeleria_AURUM_Backend/src/main/java/com/sistemapapeleria/Sistema_Papeleria_AURUM_Backend/Entidades/Inventario.java
package com.sistemapapeleria.Sistema_Papeleria_AURUM_Backend.Entidades;

import jakarta.persistence.*;
import java.util.Date;

@Entity
public class Inventario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "producto_id", referencedColumnName = "id", nullable = false)
    private Productos producto;

    private String categoria;
    private String nombreProducto;
    private int cantidadProducto;
    private Date fechaIngreso;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Productos getProducto() {
        return producto;
    }

    public void setProducto(Productos producto) {
        this.producto = producto;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public String getNombreProducto() {
        return nombreProducto;
    }

    public void setNombreProducto(String nombreProducto) {
        this.nombreProducto = nombreProducto;
    }

    public int getCantidadProducto() {
        return cantidadProducto;
    }

    public void setCantidadProducto(int cantidadProducto) {
        this.cantidadProducto = cantidadProducto;
    }

    public Date getFechaIngreso() {
        return fechaIngreso;
    }

    public void setFechaIngreso(Date fechaIngreso) {
        this.fechaIngreso = fechaIngreso;
    }
}
